spec/features/defence_request_management_spec.rb

```diff
@@ -0,0 +1,197 @@
+        expect(page).to_not have_content 'Bobby Bob Smithson'
+        expect(page).to have_field 'Solicitor Name', with: 'Bobson Smith'
+        expect(page).to have_field 'Solicitor Firm', with: 'Kreiger LLC'
+        expect(page).to have_field 'Phone Number', with: '248.412.8095'
+      end
+
+      scenario 'performing multiple own solicitor searches', js: true do
+        stub_solicitor_search_for_bob_smith
+        stub_solicitor_search_for_barry_jones
+
+        visit root_path
+        click_link 'New Defence Request'
+        choose 'Own'
+        fill_in 'q', with: "Bob Smith"
+        click_button 'Search'
+        expect(page).to have_content 'Bobson Smith'
+
+        fill_in 'q', with: "Barry Jones"
+        click_button 'Search'
+
+        expect(page).to_not have_content 'Bobson Smith'
+        expect(page).to have_content 'Barry Jones'
+      end
+
+      scenario "searching for someone who doesn't exist", js: true do
+        stub_solicitor_search_for_mystery_man
+
+        visit root_path
+        click_link 'New Defence Request'
+        choose 'Own'
+        fill_in 'q', with: "Mystery Man"
+        click_button 'Search'
+
+        expect(page).to have_content 'No results found'
+      end
+
+      scenario "toggling duty or own", js: true do
+        stub_solicitor_search_for_bob_smith
+
+        visit root_path
+        click_link 'New Defence Request'
+        choose 'Own'
+        fill_in 'q', with: "Bob Smith"
+        click_button 'Search'
+        click_link 'Bobson Smith'
+        choose 'Duty'
+
+        expect(page).to have_field 'Solicitor Name', with: "", disabled: true
+        expect(page).to have_field 'Solicitor Firm', with: "", disabled: true
+        expect(page).to have_field 'Scheme', with: "No Scheme", disabled: false
+        expect(page).to_not have_content 'Bobson Smith'
+
+        choose 'Own'
+        expect(page).to have_field 'Scheme', with: "No Scheme", disabled: true
+        expect(page).to have_field 'q', with: ''
+      end
+    end
+
+  end
+
+  context 'Edit' do
+    context 'as cso' do
+
+      before :each do
+        create_cso_and_login
+      end
+
+      let!(:dr_1) { create(:defence_request) }
+
+      scenario 'editing a DR' do
+        visit root_path
+        within "#defence_request_#{dr_1.id}" do
+          click_link 'edit'
+        end
+        expect(page).to have_content ('Edit defence request')
+
+        within '.edit_defence_request' do
+          within '.details' do
+            fill_in 'Solicitor Name', with: 'Dave Smith'
+            fill_in 'Solicitor Firm', with: 'Broken Solicitors'
+            fill_in 'Phone Number', with: '0207 284 9999'
+            fill_in 'Custody Number', with: '#CUST-9876'
+            fill_in 'Allegations', with: 'BadMurder'
+            select('10', from: 'defence_request_time_of_arrival_4i')
+            select('00', from: 'defence_request_time_of_arrival_5i')
+          end
+
+          within '.detainee' do
```

Comment:
Do we need nested `within` blocks - makes the tests dependent on the HTML design